arelle/FileSource.py

```diff
@@ -107,10 +107,8 @@ def __init__(self, url, cntlr=None, checkIfXmlIsEis=False):
         self.url = str(url)  # allow either string or FileNamedStringIO
         self.baseIsHttp = isHttpUrl(self.url)
         self.cntlr = cntlr
-        self.type = self.url.lower()[-7:]
+        self.type = self._determineType()
         self.isTarGz = self.type == ".tar.gz"
```

Comment:
Can we use [`tarfile.is_tarfile()`](https://docs.python.org/3/library/tarfile.html#tarfile.is_tarfile)?

Reply:
I added a commit to use that method.